SSA/Experimental/Bits/Fast/Tactic.lean

```diff
@@ -67,60 +67,24 @@ def termNat (n : Nat) : _root_.Term :=
   | 0 => Term.zero
   | x + 1 => Term.incr (termNat x)
 
-def incrBit (w n : Nat) :  BitStream.ofBitVec (BitVec.ofNat w n.succ) = (BitStream.ofBitVec (BitVec.ofNat w n)).incr := by
-  sorry
+theorem termNat_correct (f : Nat → BitStream) (w n : Nat) : BitStream.EqualUpTo w (BitStream.ofBitVec (BitVec.ofNat w n)) ((termNat n).eval f) := by
+  induction' n with n ih
+  · intros _ _
+    simp only [Term.eval, termNat, BitStream.zero_eq, Bool.ite_eq_false_distrib, BitVec.getLsb_zero, BitVec.msb_zero, ite_self]
+  · simp only [Term.eval, termNat, ← Nat.succ_eq_add_one]
+    trans (BitStream.ofBitVec (BitVec.ofNat w n)).incr
+    exact BitStream.ofBitVec_incr
+    exact BitStream.incr_congr ih
 
-def termNatCorrect (f : Nat → BitStream) (w n : Nat) :  BitStream.ofBitVec (BitVec.ofNat w n) = (termNat n).eval f := by
-  induction n
-  ext i
-  simp only [Term.eval ,termNat, BitStream.zero_eq, Bool.ite_eq_false_distrib, BitVec.getLsb_zero, BitVec.msb_zero, ite_self]
-  rename_i n ff
-  simp only [Term.eval ,termNat, ← ff]
-  exact incrBit w n
 
+def quoteThm (qMapIndexToFVar : Q(Nat → BitStream)) (w : Q(Nat)) (nat: Nat) :
+  Q(@BitStream.EqualUpTo $w (BitStream.ofBitVec (@BitVec.ofNat $w $nat)) (@Term.eval (termNat $nat) $qMapIndexToFVar)) := q(termNat_correct $qMapIndexToFVar $w $nat)
```

Comment:
Quick note: We should eventually un-QQ our code, to be in line with Lean core standards.

Reply:
You are probably correct, but is there any alternative to writing

```lean
.app (.app (.app (.app (.....))))
```
?